graph-gateway/src/client_query.rs

```diff
@@ -113,13 +111,19 @@ pub async fn handle_query(
     tracing::info!(deployments = ?deployments.iter().map(|d| d.id).collect::<Vec<_>>());
 
     // Check authorization for the resolved deployments
-    if !auth.are_deployments_authorized(&deployments) {
+    let resolved_deployments_ids = deployments.iter().map(|d| d.id).collect::<Vec<_>>();
+    if !auth.are_deployments_authorized(&resolved_deployments_ids) {
         return Err(Error::Auth(anyhow::anyhow!(
             "deployment not authorized by user"
         )));
     }
 
-    if !auth.are_subgraphs_authorized(&deployments) {
+    let resolved_subgraph_ids = deployments
+        .iter()
+        .flat_map(|d| d.subgraphs.iter().copied())
+        .unique()
+        .collect::<Vec<_>>();
```

Comment:
This is necessary to keep compatibility with the new decoupled module. It will be simplified once the new `network` module PR is merged.